### ISSUE TITLE
Chart rendering errors if "axis.incrs" return-value too small

### ISSUE THREAD

**Issue by Venryx** [leeoniya/uPlot#360]
Demo showing error: https://codepen.io/Venryx/pen/eYzLjLx

The chart fails to render when the line setting a custom (small) `axis.incrs` value is not commented.

If dev-tools is open when the chart is rendering, the error hits on this line: https://github.com/leeoniya/uPlot/blob/848324500fa49da0f87bce75344c39c7a996cad1/src/uPlot.js#L1338

It errors there, because the return value of `getIncrSpace` is null. It is null because the subsequent call to `findIncr` returns null.

Here is the code for `fincIncr`: https://github.com/leeoniya/uPlot/blob/848324500fa49da0f87bce75344c39c7a996cad1/src/uPlot.js#L194-L207

The problem is that the user-specified `axis.incrs` might be too small -- there might not be any values that allow the `if (space >= minSpace && minDec + incrDec < 17)` check to pass, meaning no return value is specified.

How to fix?

I'm not sure. But in any case, there should be a more informative error message. I've been hitting this issue since starting use of the library (using some workarounds to avoid), but only now discovered the cause.

The library should either be made to avoid errors for small `incrs` values, or it should give a clear error message saying what the issue is (and how to determine the minimum valid `incrs` value).

**Comment by leeoniya** [comment]
i'll make findIncr return `[0, 0]`, so tge axis renderer can bail out the same way as when getIncrSpace does:

https://github.com/leeoniya/uPlot/blob/master/src/uPlot.js#L1253-L1267

**Comment by Venryx** [comment]
In my app (not the demo), the `fullDim`/`dim` value is not 0, rather it's just very small (37).

The error occurs for this small value as well, so a `if (dim <= 0) return [0, 0];` fix would not work for my app's issue hit. (though it may solve it for the demo)

The reason the `dim` value is so small in my app, is that I start the chart out very small (100x100), prior to using a resize-observer to get the true size of the container. I've fixed it temporarily by having the start size begin at 700x700, but it seems like even if someone starts it out at 100x100, they should at least get an informative error message rather than an ambiguous crash.

(In my app, it doesn't just error/stop-rendering, it completely crashes the process, so that I can't even see what error occurred. This is what made debugging so hard; I had to add tons of log calls inside uplot's code until I narrowed down where the crash was coming from. This is unexpected behavior based just on changing the size of the width/height sent to uplot.)

**Comment by Venryx** [comment]
For the demo, here are the values at the point where `findIncr` fails to give a return value:
![](https://i.imgur.com/sE7XgbG.png)

As seen, even when `dim` is an okay value (433), the function still fails to return a value.

So basically: Whatever fix is used, it will have to account for the case screenshotted above, not just when `dim <= 0`.

**Comment by Venryx** [comment]
> i'll make findIncr return [0, 0], so the axis renderer can bail out the same way as when getIncrSpace does:

Oh, perhaps I misinterpreted. I thought you meant you'd be using an `if (dim <= 0) return [0, 0];` fix (which wouldn't fix the issue in the demo, or my app), but perhaps you meant that you'd have the function `return [0, 0];` as its last line, as a generic fallback.

I tested that fix, and it worked for the demo -- though not for my app.

However, that implies that my app's usage triggers some other error beyond the one reproduced in the demo above. So I will pursue that bug in another issue (once I figure out that second cause).
